libs/containers/User.lua

```diff
@@ -57,6 +62,18 @@ function User:getDefaultAvatarURL(size)
 	end
 end
 
+--[=[
+@m hasBadge
+@t mem
+@p badge User-Flag-Resolvable
+@r boolean
+@d Indicates whether the user has the badge given.
+]=]
+function User:hasBadge(badge)
+	badge = Resolver.userFlag(badge)
+	return band(self._public_flags or 0, badge) == badge
```

Comment:
Note that this method works differently compared to something like `Message:hasFlag()`, which checks for the presence of any of the provided flags. Might want to change it to how Message does it, or vise versa.